### DISCUSSION TITLE
What is the purpose of 'build/importTypescript.js'?

### DISCUSSION THREAD

**Discussion by wirekang** [microsoft/monaco-editor#4231] [Q&A]
I'm trying to understand why there is a https://github.com/microsoft/monaco-editor/blob/main/src/language/typescript/lib/typescriptServices.js file in monaco-editor. Afaik, Typescript works well in web browser without any modification. I tried to read every related files but couldn't find out.

**Comment by hediet** [comment]
The worker imports it: https://github.com/microsoft/monaco-editor/blob/843ff9b9d704de0960e145108c22be73f5d77c4e/src/language/typescript/tsWorker.ts#L6

**Reply by wirekang** [reply]
I mean, why worker imports the copy of typescript package instead of original in 'node_modules'?

**Reply by hediet** [reply]
I think the file might not be identical, there is a script that copies it over and does some patching.

**Reply by wirekang** [reply]
Thank you for the information 👍🏼 but I'm still wondering the purpose of the script which mocks CommonJS module and aliases namespace-related variables.
Did Typescript only support CJS module a long time ago?
